cleverhans/future/torch/utils.py

```diff
@@ -21,22 +21,7 @@ def clip_eta(eta, norm, eps):
   if norm == np.inf:
     eta = torch.clamp(eta, -eps, eps)
   else:
-    if norm == 1:
-      raise NotImplementedError("L1 clip is not implemented.")
-      norm = torch.max(
-          avoid_zero_div,
-          torch.sum(torch.abs(eta), dim=reduc_ind, keepdim=True)
-      )
-    elif norm == 2:
-      norm = torch.sqrt(torch.max(
-          avoid_zero_div,
-          torch.sum(eta ** 2, dim=reduc_ind, keepdim=True)
-      ))
-    factor = torch.min(
-        torch.tensor(1., dtype=eta.dtype, device=eta.device),
-        eps / norm
-        )
-    eta *= factor
+    eta = torch.renorm(eta, p=norm, dim=0, maxnorm=eps)
```

Comment:
Does this avoid div by zero? This is important for numerical stability in our case.

Reply:
In what case do you imagine div by zero could be an issue? I've tested and it works fine if the norm of eta is zero and fine if eps is 0